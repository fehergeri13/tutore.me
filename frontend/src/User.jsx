import React from 'react';


import "./login.less"
import Header from "./Header";

export default class App extends React.Component {
    render() {
        return <div className="userpage">
            <Header/>

            <div className="left">
                <img src="" alt=""/>

                <div className="data">
                    <div className="name">NÃ©v</div>
                    <div className="email">Email</div>
                    <div className="address">Lakhely</div>
                </div>

                <div className="feedbacks">
                    <div className="feedback">
                        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. At consectetur cupiditate dolores doloribus est expedita fuga ipsa ipsam iure, iusto labore neque, officia porro praesentium quas saepe sunt vitae, voluptatum.</p>
                    </div>
                    <div className="feedback">
                        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. At consectetur cupiditate dolores doloribus est expedita fuga ipsa ipsam iure, iusto labore neque, officia porro praesentium quas saepe sunt vitae, voluptatum.</p>
                    </div>
                </div>
            </div>

            <div className="right">

            </div>
        </div>;
    }
}
