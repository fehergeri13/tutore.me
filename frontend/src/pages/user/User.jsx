import React from 'react';
import {inject, observer} from 'mobx-react';
import {observable, toJS} from 'mobx';

import "./user.less"
import PostItem from "../../components/post/PostItem";

@inject('model')
@observer
export default class User extends React.Component {

    @observable.ref user = undefined;
    @observable.shallow userPosts = [];

    async componentDidMount() {
        const userId = this.props.model.auth.userId;
        this.user = await this.props.model.rest.getUser(userId);
        console.log('this.user', toJS(this.user));

        const response = await this.props.model.rest.getPosts();
        const filteredPosts = response.posts.filter(post => post.userId === userId);
        this.userPosts.replace(filteredPosts);

        console.log('this.userPosts', toJS(this.userPosts));
    }

    renderPosts() {
        if (this.userPosts.length === 0) {
            return <h2>Nincs hirdetés</h2>
        }

        return <ul>
            {this.userPosts.map(post => <li key={post.id}><PostItem post={post} isControl={true}/></li>)}
        </ul>;
    }

    render() {
        if (this.user === undefined) {
            return null;
        }

        return <div className="user-page">
            <div className="profile">
                <h2>Profilom</h2>
                <div className="data">
                    <div className="firstName">Vezetéknév: <span>{this.user.firstName}</span></div>
                    <div className="lastName">Keresztnév: <span>{this.user.lastName}</span></div>
                    <div className="email">Email: <span>{this.user.email}</span></div>
                </div>

                <div className="feedbacks">
                    <h2>Rólam írták</h2>

                    <ul>
                        <li>
                            <p>Nagyon király, 10/10</p>
                            <div className="stars">★★★★★</div>
                        </li>
                        <li>
                            <p>Egész jó!</p>
                            <div className="stars">★★★★☆</div>
                        </li>
                    </ul>
                </div>
            </div>

            <div className="userfeed">
                <h2>Hirdetéseim</h2>
                {this.renderPosts()}
            </div>
        </div>;
    }
}
