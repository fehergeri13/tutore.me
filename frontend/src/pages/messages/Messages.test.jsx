import React from 'react';
import renderer from 'react-test-renderer';
import User from "./Messages";

describe('User', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <User />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
