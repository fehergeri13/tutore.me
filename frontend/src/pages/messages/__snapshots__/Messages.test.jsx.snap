// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User renders correctly 1`] = `
<div
  className="messages-page"
>
  <div
    className="header"
  >
    <h1>
      tutore.me
    </h1>
    <div
      className="nav"
    >
      <ul
        className="links"
      >
        <li>
          <a
            href="#"
          >
            Hirdetések
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Új hirdetés
          </a>
        </li>
      </ul>
    </div>
    <div
      className="user"
    >
      <ul
        className="links"
      >
        <li>
          <a
            href="#"
          >
            Üzenetek
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Profilom
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Kilépés
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="left"
  >
    <ul
      className="conservations"
    >
      <li>
        <img />
        <div
          className="name"
        >
          Fülöp Farkas
        </div>
        <div
          className="last-message"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad alias amet asperiores 
        </div>
      </li>
      <li>
        <img />
        <div
          className="name"
        >
          Gipsz Józsi
        </div>
        <div
          className="last-message"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad alias amet asperiores 
        </div>
      </li>
      <li>
        <img />
        <div
          className="name"
        >
          Lellei Lilla
        </div>
        <div
          className="last-message"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad alias amet asperiores 
        </div>
      </li>
    </ul>
  </div>
  <div
    className="right"
  >
    <ul
      className="messages"
    >
      <li
        className="sent"
      >
        <span>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad alias amet asperiores 
        </span>
      </li>
      <li
        className="received"
      >
        <span>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ad alias amet asperiores 
        </span>
      </li>
      <li
        className="received"
      >
        <span>
          :D
        </span>
      </li>
      <li
        className="sent"
      >
        <span>
          Hablaka dabla abrak abra abbra
        </span>
      </li>
      <li
        className="sent"
      >
        <span>
          Wtf?
        </span>
      </li>
      <li
        className="sent"
      >
        <span>
          Ok
        </span>
      </li>
    </ul>
  </div>
</div>
`;
